server/server.go

```diff
@@ -0,0 +1,83 @@
+package main
+
+import (
+	"log"
+	"net"
+	"net/rpc"
+	"sync"
+
+	"github.com/lukedmor/conduit_tiny/common"
+)
+
+type Provider struct {
+	cs *conduitServer
+}
+
+func (p *Provider) Join(a *common.ProviderJoinLeaveArgs, reply *common.Nothing) error {
+	p.cs.providers[a.Addr] = true
+	return nil
+}
+
+func (p *Provider) Leave(a *common.ProviderJoinLeaveArgs, reply *common.Nothing) error {
+	delete(p.cs.providers, a.Addr)
+	return nil
+}
+
+type Client struct {
+	cs *conduitServer
+}
+
```

Comment:
Is there a method that checks the status of a server? If not, should we? And if so, what parameters should be checked eg. # of clients, # of providers, on/off, etc.?